feed-rs/src/xml/mod.rs

```diff
@@ -483,32 +490,36 @@ impl XmlEvent {
     // Creates a new event corresponding to an XML end-tag
     fn end<R: BufRead>(event: &BytesEnd, reader: &Reader<R>) -> XmlEvent {
         // Parse the name
-        let name = XmlEvent::parse_name(event.name(), reader);
+        let name = XmlEvent::parse_name(event.name().as_ref(), reader);
 
         XmlEvent::End { name }
     }
 
     // Extracts the element name, dropping the namespace prefix if present
     fn parse_name<R: BufRead>(bytes: &[u8], reader: &Reader<R>) -> String {
-        reader.decode(bytes).split(':').rev().next().unwrap_or("").into()
+        reader
+            .decoder()
+            .decode(bytes)
+            .ok()
```

Comment:
This drops any errors from `.decode(bytes)` which I believe was effectively the old behaviour.